pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -103,9 +101,31 @@ func newPodNIC(vmi *v1.VirtualMachineInstance, network *v1.Network, handler netd
 		podInterfaceName: podInterfaceName,
 		iface:            correspondingNetworkIface,
 		launcherPID:      launcherPID,
+		dhcpConfigurator: dhcpConfigurator,
 	}, nil
 }
 
+func newBridgeDhcpConfigurator(iface v1.Interface, vmi v1.VirtualMachineInstance, podInterfaceName string, launcherPID *int, handler netdriver.NetworkHandler, cacheFactory cache.InterfaceCacheFactory) *dhcpconfigurator.DhcpConfigurator {
+	fakeBridgeIP, err := getFakeBridgeIP(iface.Name, vmi.Spec.Domain.Devices.Interfaces)
+	if err != nil {
+		return nil
+	}
+	fakeServerAddr, err := netlink.ParseAddr(fakeBridgeIP)
+	if err != nil || fakeServerAddr == nil {
+		return nil
+	}
+	bridgeIfaceName := fmt.Sprintf("k6t-%s", podInterfaceName)
+	dhcpServer := dhcpimpl.NewBridgeDhcpServer(bridgeIfaceName, *fakeServerAddr, handler, &vmi)
+	dhcpConfigurator := dhcpconfigurator.NewDhcpConfigurator(cacheFactory, getPIDString(launcherPID), dhcpServer)
+	return &dhcpConfigurator
+}
+
+func newMasqueradeDhcpConfigurator(podInterfaceName string, launcherPID *int, handler netdriver.NetworkHandler, cacheFactory cache.InterfaceCacheFactory) dhcpconfigurator.DhcpConfigurator {
+	bridgeIfaceName := fmt.Sprintf("k6t-%s", podInterfaceName)
+	dhcpServer := dhcpimpl.NewMasqueradeDhcpServer(bridgeIfaceName, handler)
```

Comment:
ditto

Reply:
The configurator has different kinds of dhcp server implementations, that receive different parameters. That's why I'm injecting the server into the configurator.

Reply:
I mean moving `bridgeIfaceName := fmt.Sprintf("k6t-%s", podInterfaceName)` inside `NewMasqueradeDhcpServer` and what we pass to the `NewMasqueradeDhcpServer` is the podInterfaceName` so we group stuff related to DHCP to the same place.

Reply:
Why would the server care about the pod interface name ? 

IMO, it should care about the name of the interface where it should advertise from. 

I can encapsulate that in a function if you prefer.